cache.go

```diff
@@ -99,20 +111,37 @@ func New(config Config) *Cache {
 		panic("Must supply a zero or positive config.MaxAge")
 	}
 
+	if config.MinAge < 0 {
```

Comment:
Might make sense to check for sane values here instead of just 0.  I've run into type-casting serialization issues in the past due to configuration issues.  e.g. `--min-age=10` means 10ns, not 10ms or 10s as one might assume.